videoanalyser/app/src/main/java/com/example/videoanalyser/MainActivity.java

```
@@ -0,0 +1,134 @@
+                    e.printStackTrace();
+                    e.getCause();
+                }
+
+                rotationOfVideo = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION);
+                List<Bitmap> frameList = mediaMetadataRetriever.getFramesAtIndex(100, 100);
+
+                for(int i=0; i<frameList.size();i++)
+                {
+                    Log.d("Rokus Logs:", "Frame id:" + i + "\tData:"+frameList.get(i).getPixel(200,200));
+                    // analysis will be done here (forking in a separate thread)
+                    //imageHolder = InputImage.fromBitmap(frameList.get(i), Integer.parseInt(rotationOfVideo));
+                    imageHolder = InputImage.fromBitmap(frameList.get(i), 0);
+                    final int finalI = i;
+                    Task<Text> result =
+                            recognizer.process(imageHolder)
+                                    .addOnSuccessListener(new OnSuccessListener<Text>() {
+                                        @Override
+                                        public void onSuccess(Text visionText) {
+                                            // Task completed successfully
+                                            // ...
+                                            results.add(visionText.getText());
+                                            Log.d("Rokus logs", "Text detected at index:" + finalI + " " + visionText.getText());
+                                        }
+                                    })
+                                    .addOnFailureListener(
+                                            new OnFailureListener() {
+                                                @Override
+                                                public void onFailure(@NonNull Exception e) {
+                                                    // Task failed with an exception
+                                                    // ...
```

Please at least log the exception, nothing worse than code failing silently and driving youself mad trying to find what the problem is.

---

Done in refactoring commit.